authors/bruno-skvorc/polkadot/en_US/final.md

```diff
@@ -0,0 +1,186 @@
+
+Both Ethereum and Polkadot separate block production from finalization. In Ethereum, block production doesn't have a special name. In Polkadot, it's called [BABE](https://research.web3.foundation/en/latest/polkadot/BABE/Babe/) - Blind Assignment for Blockchain Extension. This separation makes sure that block production can continue without finalization, theoretically supporting blockchain growth even with a severely reduced number of validators. Once enough are back online, the finalization process can finalize the blocks starting from the last finalized one.
+
+In Ethereum, every 64th block is taken into account for finalization, and if 2/3 of validators agree (attest to) that this so called _checkpoint_ is valid, then it is _justified_. If they do this for the very next checkpoint too, or the one after it, then the checkpoint is deemed _finalized_ and can be considered irreversible. It qualifies for use by light clients and similar software which isn't interested in syncing with the entire chain's history.
+
+Polkadot is similar but the 64 slot gap isn't there. Instead, as soon as 2/3 of validators agree on a block produced by BABE according to GRANDPA rules, that block and all those before it in the same chain are immediately finalized. GRANDPA - GHOST-based Recursive ANcestor Deriving Prefix Agreement - looks at chain forks with the closest finalized block and decides which blocks to finalize based on that. A big advantage of GRANDPA is that it can finalize whole chains at once, not just block by block. This comes in handy if the validator count has been dramatically reduced (major outage perhaps). Under BABE, blocks will keep being produced, but won't be finalized because of a lack of validators. Once enough validators are back online, an entire history of blocks is finalized at once.
+
+BABE assigns a "Primary" block producer based on the VRF (normal BABE - see Randomness below) and a "Secondary" based on the following equation: `blake2_256(epoch_randomness ++ slot_number) % authorities_len`. When a block producer wins the VRF, it broadcasts a message to say that it's the primary. If no primary messages are received within some window (>>WHAT WINDOW AND HOW TO IMMUNIZE AGAINST LATENCY?), then the Secondary assignee produces the block.  With this backup in place, Polkadot slots are effectively never without blocks, with a caveat - when choosing forks, the chain with more _primary_ blocks is prioritized, so it might happen that this chain has empty slots from the VRF process.
```

Comment:
You could probably check with Alistair for an answer to this.

In fact, might be a good idea to run this article past him as well (after making our fixes) before publishing.